src/components/EnvironmentOverride/DeploymentTemplateOverrideForm.tsx

```diff
@@ -140,7 +189,36 @@ export default function DeploymentTemplateOverrideForm({
             toggleSaveChangesModal()
             return
         }
+        if (!isSuperAdmin) setSaveEligibleChangesCb(!saveEligibleChangesCb)
+    }
+
+    const handleSaveChanges = (e) => {
+        e.preventDefault()
+        handleSubmit(false)
+    }
+    const handleChangeCheckbox = () => {
+        if (!saveEligibleChangesCb) {
+            checkForSaveAsDraft()
+        } else {
+            if (state.showSaveChangesModal) toggleSaveChangesModal()
+            else {
+                setSaveEligibleChangesCb(!saveEligibleChangesCb)
+            }
+        }
+    }
 
+    const checkForProtectedLockedChanges = async () => {
+        const data = prepareDataToSaveDraft()
+        const action = data['id'] > 0 ? 2 : 1
+        const requestPayload = {
+            appId: Number(appId),
+            envId: Number(envId),
+            action,
+            data: JSON.stringify (data),
+        }
+        return await getIfLockedConfigProtected(requestPayload)
```

Comment:
We should add try catch for error handling

Reply:
The error handling is there in function from where the function is being called, So I think it will be handled in parent function
Otherwise if I do error handle here also then flow after this function call be executed in parent function, if any better solution please suggest